packages/hash/shared/src/entityStorePlugin.ts

```diff
@@ -173,15 +179,15 @@ const updateBlockEntity = (
   draftBlockId: string,
   targetEntity: EntityStoreType,
 ) => {
-  let targetDraftEntity = draftEntityForEntityId(
+  let targetDraftEntity = getDraftEntityFromEntityId(
     draftEntityStore,
     draftBlockId,
   );
 
   // Add target entity to draft store if it is not
   // present there
   if (!targetDraftEntity) {
-    const targetEntityDraftId = newDraftId();
+    const targetEntityDraftId = createDraftIdForEntity(targetEntity.entityId);
```

Comment:
We seem to definitely have a `targetEntity` here, so I assume we should be using its real draftId rather than a `fake-` one

Reply:
Not exactly. The target entity here is what the user selects from the LoadEntity menu. There's a possibility the entity will not be present in the draft store and that's why we have the conditional.  We try to pull it from the draft store in line 182. If it's present, we go ahead working with it. If it's not, we have to add it to the draft store before continuing.

Reply:
Right, but why does the `draftId` have to be `fake-(uuid)`? That's what passing `null` does here.

I thought the reason we had `fake-` ids was because the entity might not have an `entityId` yet to use to make `draft-(entityId)`, i.e. if we just created the entity - but this one does have an `entityId`? How do we end up with an entity in the list without an `entityId`?

The type of `EntityStoreType["entityId"]` is string or null, so `targetEntity.entityId` should be string or null here anyway - unless we need to fix the type?

Reply:
Also, `targetDraftEntity` is the draft entity of the block we want to change, isn't it? Since it's looked up by `draftBlockId` in line 182. But this is dealing with the child entity we want to change the block to.

Is `targetDraftEntity` supposed to be looked up using `targetEntity` instead?

Reply:
I assumed that we use `fake-(uuid)` for any new entity we add to draft store from the client side. Re-thinking about  it, you are correct since we already have the entity we should use `draft-(entityId)`. Makes it easy to reason about. Both approaches would work if we always use `getDraftEntityFromEntityId` to fetch the draft entity anyway. Let's stick to only using `fake-(uuid)` when dealing with new entities without entityId like you proposed

Right now we get an entity in the list without an entityId when we create a new draft block.